cmrf_views/src/Plugin/views/query/API4.php

```diff
@@ -0,0 +1,525 @@
+        }
+      }
+
+      // Set the parameters from the dataset params options.
+      if (!empty($dataset_params)) {
+        // Replace tokens recursively.
+        array_walk_recursive($dataset_params, ['\Drupal\cmrf_views\CMRFViews', 'tokenReplace']);
+        $parameters = array_merge($parameters, $dataset_params);
+      }
+
+      // Count options.
+      $options['cache'] = empty($view->query->options['cache']) ? NULL : $view->query->options['cache'];
+
+      // Count API call.
+      $count_parameters = $parameters;
+      $count_parameters['select'] = ['row_count'];
+      $call = $this->core->createCall($connector, $api_entity, $api_action, $count_parameters, $options, NULL, $api_version);
+      $this->core->executeCall($call);
+      if ($call->getStatus() == Call::STATUS_DONE) {
+        $result = $call->getReply();
+        $view->getPager()->total_items = $result['countMatched'] ?? $result['count'];
+        $view->total_rows              = $result['countMatched'] ?? $result['count'];
+      }
+
+      // Update pager.
+      $view->getPager()->updatePageInfo();
+
+      // TODO: verify views cache.
+      $parameters['limit']  = $view->getPager()->getItemsPerPage();
+      $parameters['offset'] = $view->getCurrentPage() * $view->getPager()->getItemsPerPage();
+
+      // View result init.
+      $view->result = [];
+
+      $field_id_alias_mapping = array_combine(
+        array_keys($view->field),
+        array_column($view->field, 'field_alias')
+      );
+
+      // Data API call.
+      $call = $this->core->createCall($connector, $api_entity, $api_action, $parameters, $options, NULL, $api_version);
+      $this->core->executeCall($call);
+      if ($call->getStatus() == Call::STATUS_DONE) {
+        $result = $call->getReply();
+        if ((!empty($result['values'])) && (is_array($result['values']))) {
+          $index = 0;
+          foreach ($result['values'] as $row) {
+            // Mandatory field for views rows.
+            $row['index'] = $index++;
+            // Add row to view result.
+            $base_result = [];
+            foreach ($row as $key => $value) {
+              if ($field_alias = self::getFieldAlias($view->storage->get('base_table'), $key)) {
+                $base_result[$field_alias] = $value;
+              }
+            }
+            $view->result[] = new CMRFViewsResultRow($base_result);
+          }
+          // Set row indices for template_preprocess_views_view_fields to be
+          // able to retrieve the values.
+          array_walk($view->result, function (ResultRow $row, $index) {
+            $row->index = $index;
+          });
+        }
+      }
+
+      // TODO: Adjust relationships for APIv4.
+      foreach ($view->relationship as $field_name => $relationship) {
+        $field_name = self::getFieldAlias($view->storage->get('base_table'), $field_name);
+        $referenced_keys = [];
+        foreach ($view->result as $row_key => &$row) {
```

Comment:
`$row_key` is unused. There's no assignment to `$row` so it does not need to be prefixed by `&`.

Reply:
It eats no bread though, but I'll remove it.